BackEndLib/Wchar.cpp

```diff
@@ -560,20 +560,25 @@ int _Wtoi(const WCHAR* wsz)
 }
 
 //*****************************************************************************
-WCHAR* _itoW(int value, WCHAR* wcz, int radix)
+WCHAR* _itoW(int value, WCHAR* wcz, int radix, int bufferLength)
 {
 	int i = 0;
 	bool bNegative = false;
 	if (value < 0) //negative?
 	{
 		value = -value;
 		bNegative = true;
+		bufferLength--; // We have one character fewers for numbers because we need negative sign
 	}
+	bufferLength--; // Make space for null termination char
 	do {
 		int val = value % radix;
 		if (val < 10)
 			buffer[i++] = (value % radix) + '0';
 		else buffer[i++] = (value % radix) - 10 + 'A';
+
+		if (i >= bufferLength) // Ran out of space in the buffer
```

Comment:
I'm not worried about speed impact from introducing this check.
(I would expect time spent doing the multiple string copies below dominates.)

However, I can't think of a use case where we'd actually want to truncate a numeric conversion because the source buffer isn't large enough.  This is a bug; we should simply resize the input string so the result fits.

Nothing wrong, per se, with keeping the check, though.  Some suggestions:
1. Add an assertion failure if we reach this point.
1a. Swap the if check for an assertion.
2. Change the break to a thrown exception.

Thoughts?

Reply:
The only reason I'd be tempted to truncate rather than crash hard would be to make it slightly more user friendly - a crashing game can make you lose a lot of progress. On the other hand once, it's fixed in a given place once it should never be a problem again. 

I think an assertion beep but gracefully trimming the excess is the most user friendly solution here.